p2p/peer.py

```diff
@@ -738,14 +739,21 @@ def add_peer(self, peer):
             self.logger.exception("Unexpected error during auth/p2p handshake with %s", remote)
         return None
 
-    async def lookup_random_node(self) -> None:
-        # This method runs in the background, so we must catch OperationCancelled here otherwise
-        # asyncio will warn that its exception was never retrieved.
-        try:
-            await self.discovery.lookup_random(self.cancel_token)
-        except OperationCancelled:
-            pass
-        self._last_lookup = time.time()
+    async def maybe_lookup_random_node(self) -> None:
+        if self._discovery_last_lookup + self._discovery_lookup_interval > time.time():
+            return
+        elif self._discovery_lookup_running.locked():
+            self.logger.debug("Node discovery lookup already in progress, not running another")
+            return
+        with (await self._discovery_lookup_running):
```

Comment:
is this equvalent to `async with self.discovery_lookup_running`?

That's what I found here: https://www.pythonsheets.com/notes/python-asyncio.html

Reply:
Looks like it: https://www.python.org/dev/peps/pep-0492/#asynchronous-context-managers-and-async-with

I'll switch to `async with`; I think that reads better